cli/src/main/java/org/jboss/as/cli/handlers/AttachmentHandler.java

```diff
@@ -0,0 +1,399 @@
+        }
+        ModelNode mn = ctx.buildRequest(op);
+        return mn;
+    }
+
+    private static class AttachmentResponseHandler implements ResponseHandler {
+
+        private final boolean save;
+        private final String targetPath;
+        private final CommandContext ctx;
+        private final boolean override;
+        private AttachmentResponseHandler(CommandContext ctx, String targetPath,
+                boolean save, boolean override) {
+            this.ctx = ctx;
+            this.targetPath = targetPath;
+            this.save = save;
+            this.override = override;
+        }
+
+        @Override
+        public void handleResponse(ModelNode step, OperationResponse response)
+                throws CommandLineException {
+            //First retrieve all uuid.
+            Set<String> uuids = getStreams(response.getResponseNode());
+            if (uuids.isEmpty()) {
+                return;
+            }
+            //Then lookup the complete data structure for a possible match.
+            Set<String> mystreams = new TreeSet<>();
+
+            // In case of a non composite operation, the headers are located
+            // inside the step ModelNode.
+            // So, although the operation wouldn't return the uuid in the result,
+            // we find the uuid in the headers.
+            // Obviously this would not work in batch mode (composite).
+            retrieveStreams(step, uuids, mystreams);
+
+            int index = 0;
+            for (String uuid : mystreams) {
+                if (save) {
+                    index = saveStream(uuid, response, index);
+                } else {
+                    displayStream(uuid, response);
+                }
+            }
+        }
+
+        private int saveStream(String uuid, OperationResponse response,
+                int index) throws CommandLineException {
+            String target = targetPath == null ? uuid : targetPath;
+            OperationResponse.StreamEntry entry = response.getInputStream(uuid);
+            File targetFile = new File(target);
+            if (!override) {
+                while (targetFile.exists()) {
+                    String name = targetFile.getName();
+                    int indexed = name.lastIndexOf("(");
+                    if (indexed > 0) {
+                        try {
+                            String num = name.substring(indexed + 1, name.length() - 1);
+                            index = Integer.valueOf(num);
+                            index += 1;
+                            name = name.substring(0, indexed);
+                        } catch (NumberFormatException ex) {
+                            // XXX OK, not a number.
+                        }
+                    } else {
+                        index += 1;
+                    }
+                    targetFile = new File(name + "(" + index + ")");
+                }
+            }
```

Comment:
If override is true and the response contains multiple streams, shouldn't the index be incremented here?